pkg/cli/workflow/list_runs.go

```diff
@@ -0,0 +1,119 @@
+
+	workflowc "github.com/fuseml/fuseml-core/gen/http/workflow/client"
+	"github.com/fuseml/fuseml-core/gen/workflow"
+	"github.com/fuseml/fuseml-core/pkg/cli/common"
+)
+
+type listRunsOptions struct {
+	common.Clients
+	global         *common.GlobalOptions
+	format         *common.FormattingOptions
+	name           string
+	codesetName    string
+	codesetProject string
+	status         string
+}
+
+func formatRunDuration(object interface{}, column string, field interface{}) string {
+	if wr, ok := object.(*workflow.WorkflowRun); ok {
+		layout := time.RFC3339
+		startTime, _ := time.Parse(layout, *wr.StartTime)
+		completionTime := time.Time{}
+		if wr.CompletionTime != nil {
+			completionTime, _ = time.Parse(layout, *wr.CompletionTime)
+		}
+		return formatted.Duration(&v1.Time{Time: startTime}, &v1.Time{Time: completionTime})
+	}
+	return ""
+}
+
+func formatRunStatus(object interface{}, column string, field interface{}) string {
+	if wr, ok := object.(*workflow.WorkflowRun); ok {
+		return formatted.ColorStatus(*wr.Status)
+	}
+	return ""
+}
+
+func formatRunWorkflowRef(object interface{}, column string, field interface{}) string {
+	if wr, ok := object.(*workflow.WorkflowRun); ok {
+		return *wr.WorkflowRef
+	}
+	return ""
+}
+
+func formatRunStartTime(object interface{}, column string, field interface{}) string {
+	if wr, ok := object.(*workflow.WorkflowRun); ok {
+		startTime, _ := time.Parse(time.RFC3339, *wr.StartTime)
+		return formatted.Age(&v1.Time{Time: startTime}, clockwork.NewRealClock())
+	}
+	return ""
+}
+
+func newListRunsOptions(o *common.GlobalOptions) (res *listRunsOptions) {
+	res = &listRunsOptions{global: o}
+	res.format = common.NewFormattingOptions(
+		[]string{"Name", "Workflow", "Started", "Duration", "Status"},
+		[]table.SortBy{{Name: "Started", Mode: table.Dsc}},
+		common.OutputFormatters{"Duration": formatRunDuration, "Status": formatRunStatus,
+			"Workflow": formatRunWorkflowRef, "Started": formatRunStartTime},
+	)
+
+	return
+}
+
+func newSubCmdListRuns(c *common.GlobalOptions) *cobra.Command {
+	o := newListRunsOptions(c)
+	cmd := &cobra.Command{
+		Use:   "list-runs [-n|--name NAME] [-c|--codeset-name CODESET_NAME] [-p|--codeset-project CODESET_PROJECT] [-s|--status STATUS]",
+		Short: "Lists one or more workflow runs",
+		Long:  `Prints a table of the most important information about workflow runs. You can filter the list by the workflow name, codeset name, codeset project or status.`,
+		Run: func(cmd *cobra.Command, args []string) {
+			common.CheckErr(o.InitializeClients(c))
+			common.CheckErr(o.validate())
+			common.CheckErr(o.run())
+		},
+		Args: cobra.ExactArgs(0),
+	}
+
+	cmd.Flags().StringVarP(&o.name, "name", "n", "", "filter workflow runs by the workflow name")
+	cmd.Flags().StringVarP(&o.codesetName, "codeset-name", "c", "", "filter workflow runs by the codeset name")
+	cmd.Flags().StringVarP(&o.codesetProject, "codeset-project", "p", "", "filter workflow runs by the codeset project")
```

Comment:
The project is not directly associated with a workflow run, but I'd like to think we'll make that happen really soon. When that happens, these two would be better named `codeset` and `project`. Basically, all workflow runs created from a codeset would "belong" to the same project as the codeset. And if you only use the codeset name here without specifying the project, you should be able to find the codeset by using the same project that the workflow run is associated with.

But these are just thoughts about the future... all is good here otherwise.